client/blocks/comment-detail/docs/comment-faker.jsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { keyBy, omit } from 'lodash';
+
+/**
+ * `CommentFaker` is a HOC to easily test the Comments Management without the necessity of real data or actions.
+ *
+ * It takes the comments passed via props (with mock data, Redux-connected, etc.) and puts them in its internal state.
+ * Then passes them and various actions (status and like setters) to the actual comment list component.
+ *
+ * Once data and actions will be completely Reduxified, it will be enough to remove the `CommentFaker` call
+ * and use the `mapDispatchToProps` actions instead, leaving this component still useful for testing purposes.
+ */
+export const CommentFaker = WrappedCommentList => class extends Component {
+	state = {
+		comments: {},
+	};
+
+	componentWillMount() {
+		if ( this.props.comments.length ) {
+			this.setState( { comments: keyBy( this.props.comments, 'ID' ) } );
+		}
+	}
```

Comment:
The reasoning here is as follow:

- If we have `props.comments` at `willMount`: we are passing them manually (like in the CommentDetail docs). Pull those and call it a day.

- If we manage to get to `willReceiveProps` with an empty `props.comments`: we are waiting for the Redux state to be populated. Once it's populated, pull the comments in state and stop checking for `props.comments` changes, because we don't need them in the faker.

Now, that check started in `willReceiveProps` to avoid performing unnecessary `setState()` and accidentally changing `state.comments`.
But yeah, of course it's not needed in `willMount`.